docs/2-fork-accountability.md

```diff
@@ -0,0 +1,163 @@
+
+The agreement property of the Tendermint consensus algorithm cannot be violated if there are at most *f* faulty processes in the system.
+What happens if more than *f* processes are faulty?
+
+A well-designed consensus protocol should provide some guarantees when this limit is exceeded. The most important guarantee is **fork accountability**. 
+
+## The problem
+
+A **fork** happens when two correct validators decide on different blocks in the same height or, by using different words, when there are two commits for different blocks at the same height. 
+In the context of the Tendermint protocol, a fork happens in height h when a quorum of messages (2f + 1 Precommit messages) are sent for different values: there are two sets of 2f + 1 Precommit messages, A and B, such that A has value v and B has value v' != v in height h.  
+
+Tendermint conensus algorithm guarantees that a fork (i.e., a violation of the agreement property) cannot happen when the number of faulty processes is less than one third.
+We address the problem of forks when there are more than one third of faulty validators:
+
+    f < num. faulty processes <= 2f
+
+In the Tendermint protocol, all processes should be accountable for their actions and faulty validators should be identified and punished according to the protocol specifications because they can "cause" the consensus to fail.
+Fork accountability aims to detect these faulty processes that "led" to a fork and punish them, for example by removing them from the validators set.
+
+We aim to give incentives to validators to behave correctly according to the protocol specifications and detect faulty validators, without mistakenly detecting correct validators. 
+ 
+## Fork reasons
+ 
+There are multiple reasons that can lead to a fork, all coming from processes that deviate from the protocol specification and not following the behavior of a correct process.
+
+We can summarize the most important ones:
+
+- A process sends multiple proposals in a round *r* for different values
+
+- A process sends a Prevote/Precommit message for an invalid value *v* in a round *r*
+
+- A process sends multiple Prevote/Precommit messages in a round *r* for different values
+
+- A process sends a Prevote/Precommit message for a value *v* despite having a lock on a different value *v'*
+
+- A process sends a Precommit for a value *v* without having received at least 2/3 Prevote messages for *v* 
+
+## Design of an accountability algorithm
+
+### Interface and properties
+
+Given the problemas stated above, we want to develop an accountability algorithm that, by analyzing the messages received from validators during the execution of the consensus instance, is able to determine which processes respected the protocol and which ones did not and led to a fork. 
+In order to prove the misbehavior of the processes that did not respect the protocol, we want to show the proof of their faultiness with respect to the Tendermint consensus protocol.
```

Comment:
I know your code already uses the term "faultiness". Another term we can use here, which is more common, would be "misbehavior". This is a matter of taste, however, so feel free to ignore this suggestion.

Reply:
I agree, it sounds better. Fixed.